src/easier68k/core/models/assembly_parameter.py

```diff
@@ -0,0 +1,219 @@
+"""
+Assembly Parameter
+
+Represents a single parameter in assembly code
+this is the new 'EAMode' class
+"""
+
+from ..enum.ea_mode import EAMode
+from ..enum.register import Register
+from ...simulator.m68k import M68K
+from ..util.conversions import to_word
+
+# should try to make this a constant only defined once
+MAX_MEMORY_LOCATION = 16777216  # 2^24
+
+
+class AssemblyParameter:
+
+    def __init__(self, mode: EAMode = EAMode.ERR, data: int = 0, length: int = 4):
+        """
+        Constructor
+
+        Checks that the mode and the data are considered to be valid, like
+        registers are within the correct bounds and addresses are not out of bounds.
+        """
+
+
+        # ensure that the values are valid
+
+        # when referencing a register, ensure that the data is within [0, 7]
+        if mode in [EAMode.DataRegisterDirect, EAMode.AddressRegisterIndirectPreDecrement,
+                    EAMode.AddressRegisterIndirect, EAMode.AddressRegisterIndirectPostIncrement,
+                    EAMode.AddressRegisterDirect]:
+            assert 0 <= data <= 7, 'The register number for this mode must be in the range [0, 7]!'
+
+        if mode in [EAMode.AbsoluteWordAddress, EAMode.AbsoluteLongAddress]:
+            # ensure that the address isn't outside the bounds of max memory location
+            # and is greater than or eq to 0
+            assert 0 <= data <= MAX_MEMORY_LOCATION, 'An absolute address must be in the bounds [0, 2^24]!'
+
+        if mode is EAMode.Immediate:
+            # ensure that the value isn't too large, cannot be larger than a long word
+            # negative values need to be converted into unsigned integers
+            assert 0 <= data <= 0xFFFFFFFF, 'Value must fit inside a long word!'
+
+        assert mode is not EAMode.ERR, 'EAMode cannot be ERROR.'
```

Comment:
You check against ERR in other code, but that condition will never happen because of this assert. I'm not sure which intention you're going for (to just raise the assertion right away or later when it's referenced in moving)

Reply:
I think if we go with the previous comment of removing ERR that'll resolve this too.